crates/cli-support/src/js/mod.rs

```diff
@@ -345,7 +345,8 @@ impl<'a> Context<'a> {
             }}
 
             const wasmInstance = (await WebAssembly.instantiate(wasmCode, imports)).instance;
-            const wasm = wasmInstance.exports;",
+            const wasm = wasmInstance.exports;
+            export const __wasm = wasm;",
```

Comment:
Note for reviewer: this is probably undesirable and at the very least should be limited to Deno.

Reply:
It didn't broke node or any of the implementations I tested so far, so I wasn't sure if deno was just being more strict than the others.

Reply:
I wasn't worried about breaking, I'm more worried about exporting anything, even if prefixed by `__`, unless its absolutely necessary. We don't want anyone starting to rely on it unless we intend to and we also don't want to occupy a slot unless necessary.

Again, there is no real issue here, we should just double-check if this is necessary.

Reply:
I understand your concerns, the generated code requires this variable both on node and on deno
```js
const ok = await cx.run(tests.map(n => wasm.__wasm[n]))
```
For some reason, without the export it says its not declared on deno, perhaps because deno uses an import:
```js
import * as wasm from "./{0}.js";
```
while node still uses a require:
```js
const wasm = require("./{0}")
```
In practice it was being "exported" for node already, intentionally or not.

Reply:
It seems like this was added for Node only in #2012: https://github.com/rustwasm/wasm-bindgen/blob/ad826519586351e1cafbd203e74e8db5b46a89a0/crates/cli-support/src/js/mod.rs#L278

I don't like it very much, but this isn't making things any worse and removing it from Node is probably out of scope for this PR.

Reply:
I'm not sure about the best approach either here or in other places, but it seems clear to me that some refactors are in order.

I don't think its just me, but when I'm just trying to fix issues, I try to avoid refactoring as much as possible to make the PR easier to review, but that strategy adds up, and its pretty clear that some refactors are in order.

Although, in this case and some others, I'm not sure the best approach is, as my point of view was quite narrow, focused on specific issues.

But I think opening issues with those changes, so that someone can pick up and provide a PR might be a very good idea.